### PR TITLE
Code-First: Backend API

### PR DESCRIPTION
**Pull request by embryCODE**
I mostly followed the instructions and kept this very simple and naive, but I couldn't help myself on a few things. I tweaked eslint a bit and I used an enum for error messages and created a ResponseDTO class. All this validation in a transaction script was making my code already spaghetti-ish!

### CODE REVIEW

File: ThePhasesOfCraftship/1_code_first/assignment/mason-embry/prisma/schema.prisma [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  url      = env("DATABASE_URL")
+}
+
+model User {
+  id        Int     @id @default(autoincrement())
+  email     String  @unique
+  username  String  @unique
+  firstName String
+  lastName  String
+  password  String  @default(uuid()) // Generates a random UUID as password
```
Comment (embryCODE) [inline ThePhasesOfCraftship/1_code_first/assignment/mason-embry/prisma/schema.prisma:22]: I saw you do this in code. This seemed easier.

File: ThePhasesOfCraftship/1_code_first/assignment/mason-embry/src/iDontKnowWhereToPutThis/responseDTO.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+export enum ErrorMessage {
```
Comment (embryCODE) [inline ThePhasesOfCraftship/1_code_first/assignment/mason-embry/src/iDontKnowWhereToPutThis/responseDTO.ts:1]: I couldn't help myself with this file. I'm also thinking of creating my own Error classes that extend Error.

File: ThePhasesOfCraftship/1_code_first/assignment/mason-embry/src/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (embryCODE) [inline ThePhasesOfCraftship/1_code_first/assignment/mason-embry/src/index.test.ts:1]: I prefer test to spec so I changed this file name

File: ThePhasesOfCraftship/1_code_first/assignment/mason-embry/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+const prisma = new PrismaClient();
+
+const app = express();
+
+app.use(express.json());
+app.use(cors());
+
+// Create a new user
+app.post('/users/new', async (req: Request, res: Response) => {
```
Comment (embryCODE) [inline ThePhasesOfCraftship/1_code_first/assignment/mason-embry/src/index.ts:18]: It hurts to keep all of this in index.ts, but I think that's what was requested at this stage.

File: ThePhasesOfCraftship/1_code_first/assignment/mason-embry/src/iDontKnowWhereToPutThis/responseDTO.ts (hunk 2 above)
Comment (AnDeVerin) [reply ThePhasesOfCraftship/1_code_first/assignment/mason-embry/src/iDontKnowWhereToPutThis/responseDTO.ts]: It seems like utilitarian functionality to me, so I'd suggest putting it in to the utils folder.

File: ThePhasesOfCraftship/1_code_first/assignment/mason-embry/src/index.ts (hunk 4 above)
Comment (AnDeVerin) [reply ThePhasesOfCraftship/1_code_first/assignment/mason-embry/src/index.ts]: I guess it's OK for the walking skeleton. :-)

**Pull request closed without merging**